### PR TITLE
Supporting attestation formats

### PR DESCRIPTION
**Pull request by jonashendrickx**
Adding support for attestation formats.

The RP can use this optional member to specify a preference regarding the [attestation](https://www.w3.org/TR/webauthn-3/#attestation) statement format used by the [authenticator](https://www.w3.org/TR/webauthn-3/#authenticator). Values SHOULD be taken from the IANA "WebAuthn Attestation Statement Format Identifiers" registry [[IANA-WebAuthn-Registries]](https://www.w3.org/TR/webauthn-3/#biblio-iana-webauthn-registries) established by [[RFC8809]](https://www.w3.org/TR/webauthn-3/#biblio-rfc8809). Values are ordered from most preferable to least preferable. This parameter is advisory and the [authenticator](https://www.w3.org/TR/webauthn-3/#authenticator) MAY use an attestation statement not enumerated in this parameter.

- https://www.w3.org/TR/webauthn-3/#dom-publickeycredentialcreationoptions-attestationformats
- https://www.w3.org/TR/webauthn-3/#dictionary-assertion-options

### CODE REVIEW

File: Src/Fido2.Models/AssertionOptions.cs [hunk 1]
```diff
@@ -54,6 +54,12 @@ public class AssertionOptions : Fido2ResponseBase
     [JsonPropertyName("hints")]
     public IReadOnlyList<PublicKeyCredentialHint> Hints { get; set; } = Array.Empty<PublicKeyCredentialHint>();
 
+    /// <summary>
+    /// This member is intended for use by Relying Parties that wish to select a preference regarding the attestation statement format used, if such an attestation is requested.
+    /// </summary>
+    [JsonPropertyName("attestationFormats")]
+    public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats = Array.Empty<AttestationStatementFormatIdentifier>();
+
```
Comment (iamcarbon) [inline Src/Fido2.Models/AssertionOptions.cs:62]: This should be a property.

```
public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats { get; set; } = [];
```

File: Src/Fido2.Models/CredentialCreateOptions.cs [hunk 2]
```diff
@@ -49,6 +49,15 @@ public sealed class CredentialCreateOptions : Fido2ResponseBase
...
     public AttestationConveyancePreference Attestation { get; set; } = AttestationConveyancePreference.None;
 
+    /// <summary>
+    /// This member is intended for use by Relying Parties that wish to select a preference regarding the attestation statement format used, if such an attestation is requested.
+    /// </summary>
+    /// <remarks>
+    /// This parameter is advisory and the authenticator MAY use an attestation statement not enumerated in this parameter.
+    /// </remarks>
+    [JsonPropertyName("attestationFormats")]
+    public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats = Array.Empty<AttestationStatementFormatIdentifier>();
```
Comment (iamcarbon) [inline Src/Fido2.Models/CredentialCreateOptions.cs:59]: This should be a property.

File: Src/Fido2.Models/Objects/AuthenticationExtensionsDevicePublicKeyInputs.cs [hunk 3]
```diff
@@ -9,5 +9,5 @@ public sealed class AuthenticationExtensionsDevicePublicKeyInputs
     public string Attestation { get; set; } = "none";
 
     [JsonPropertyName("attestationFormats")]
-    public string[] AttestationFormats { get; set; } = Array.Empty<string>();
+    public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats { get; set; } = Array.Empty<AttestationStatementFormatIdentifier>();
```
Comment (iamcarbon) [inline Src/Fido2.Models/Objects/AuthenticationExtensionsDevicePublicKeyInputs.cs:12]: Consider using a collection expression.

```csharp
 public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats { get; set; } = [];
```

File: Src/Fido2.Models/AssertionOptions.cs (hunk 1 above)
Comment (jonashendrickx) [reply Src/Fido2.Models/AssertionOptions.cs]: Thank you @iamcarbon

File: Src/Fido2.Models/AssertionOptions.cs [hunk 4]
```diff
@@ -54,6 +54,12 @@ public class AssertionOptions
     [JsonPropertyName("hints")]
     public IReadOnlyList<PublicKeyCredentialHint> Hints { get; set; } = Array.Empty<PublicKeyCredentialHint>();
 
+    /// <summary>
+    /// This member is intended for use by Relying Parties that wish to select a preference regarding the attestation statement format used, if such an attestation is requested.
+    /// </summary>
+    [JsonPropertyName("attestationFormats")]
+    public IReadOnlyList<AttestationStatementFormatIdentifier> AttestationFormats { get; set; } = [];
```
Comment (iamcarbon) [inline Src/Fido2.Models/AssertionOptions.cs:61]: Can we extend this description to include:

Values are ordered from most preferable to least preferable. This parameter is advisory and the authenticator MAY use an attestation statement not enumerated in this parameter.

**Pull request merged**